### DISCUSSION TITLE
Market open and close with a different TZ

### DISCUSSION THREAD

**Discussion by saoudm** [gerrymanoim/exchange_calendars#83] [Ideas]
Hello,

Would it possible to use datetime.time.tzinfo of ExchangeCalendar.open_times and ExchangeCalendar.close_times instead of ExchangeCalendar.tz to create a custom calendar that has different timezones for market opens and closes (and breaks)

Thanks

**Comment by maread99** [comment]
Hi @saoudm, I just both +1'ed and -1'ed your comment - neither of which were intended! (there doesn't seem to be an undo option...).

With regards to your question, without looking through the implementation I suspect that it could be done. Maybe along the lines of 'if the start_date of `open_times` / `close_times` has a tz then use it, otherwise use ExchangeCalendar.tz'. Although a little thought might have to be given to how to handle a specific tz when the start_date is None.

However, I wonder whether what you're asking for is too niche and abstract to warrant a change to the stardard implementation. It would be interesting to see if other users support this functionality.

Thanks for the question!

**Reply by saoudm** [reply]
Hello,

Looking at the code, unfortunately it looks like the open_times tz is not used in the function pandas_utils:days_at_time.

My suggested change is instead of  `return (days + delta).tz_localize(tz).tz_convert(UTC)`  we can use the time.tzinfo  `  return (days + delta).tz_localize(t.tzinfo or tz).tz_convert(UTC)`
